training_pipeline/pipeline/configs.py

```diff
@@ -42,6 +43,59 @@
 EXAMPLE_GEN_BEAM_ARGS = None
 TRANSFORM_BEAM_ARGS = None
 
+"""
+EVAL_CONFIGS is to configuration for the Evaluator component
+to define how it is going to evalua the model performance.
+
+tfma.ModelSpec
+    signature_name is one of the signature in the SavedModel from
+    Trainer component, and it will be used to make predictions on 
+    given data. preprocessing_function_names allows us to include
+    a set of transformation(preprocessing) signatures in the Saved
+    Model from Trainer component. 
+
+    label_key and prediction_key will be used to compare the ground
+    truth and prediction results.
+
+slicing_specs
+    we use the entire dataset to evaluate the model performance. If
+    you want to evaluate the model based on different slices of data
+    set, you should prepare TFRecords to have multiple features which
+    of each corresponds to each slices(or categories), then write the
+    slicing_specs options accordingly.
+"""
+EVAL_CONFIGS = tfma.EvalConfig(
+    model_specs=[
+        tfma.ModelSpec(
+            signature_name="from_examples",
+            preprocessing_function_names=["transform_features"],
+            label_key="label_xf",
+            prediction_key="label_xf",
+        )
+    ],
+    slicing_specs=[tfma.SlicingSpec()],
+    metrics_specs=[
+        tfma.MetricsSpec(
+            metrics=[
+                tfma.MetricConfig(
+                    class_name="SparseCategoricalAccuracy",
+                    threshold=tfma.MetricThreshold(
+                        value_threshold=tfma.GenericValueThreshold(
+                            lower_bound={"value": 0.55}
+                        ),
+                        # Change threshold will be ignored if there is no
+                        # baseline model resolved from MLMD (first run).
+                        change_threshold=tfma.GenericChangeThreshold(
+                            direction=tfma.MetricDirection.HIGHER_IS_BETTER,
+                            absolute={"value": -1e-3},
+                        ),
```

Comment:
What do these denote?

Reply:
We can specify two models in the `Evaluator` component. One is the currently trained model, and the other one is the best model currently deployed(retrieved from the Artifact Store). 

`tfma.GenericChangeThreshold` let us to define the threshold by how much the currently trained model should be better than the previous model to replace it.
- we are evaluating the model based on `accuracy`, and higher score means better, so `tfma.MetricDirection.HIGHER_IS_BETTER` is set.

Reply:
Makes a lot of sense. Thanks!